linera-service/src/cli_wrappers/kubectl.rs

```diff
@@ -6,23 +6,44 @@ use anyhow::{Context, Result};
 use tokio::process::{Child, Command};
 
 pub struct KubectlInstance {
-    pub port_forward_child: Option<Child>,
+    pub port_forward_children: Option<Vec<Child>>,
 }
 
 impl KubectlInstance {
-    pub fn new(port_forward_child: Option<Child>) -> Self {
-        Self { port_forward_child }
+    pub fn new(port_forward_children: Option<Vec<Child>>) -> Self {
+        Self {
+            port_forward_children,
+        }
+    }
+
+    // If there are multiple clusters, make sure you always properly set the context
+    // before doing kubectl commands, otherwise you might be querying the wrong cluster
+    pub async fn config_use_context(&mut self, cluster_id: u32) -> Result<()> {
```

Comment:
This sets the context for the client at the user level right?

So if I connected with my kubectl to cluster A, then I run `linera net up --k8s`, the I resume work with kubectl, it will be connected to a different cluster right?

So we either need a way to set the context just for the command - or find a way to reset it after we are done.

Furthermore, if you are going to be re-using the context it feels like this should be a field of `KubectlInstance`. Something like:


```rust
struct KubectlInstance {
    root_cmd: Command
}

impl KubectlInstance {
  fn new(id: u32) -> Self {
      Self {
         root_cmd: Command::new("kubectl").arg("context").arg(id)
      }
  }
}
```

And then re-use `root_cmd` for subsequent commands like port-forwarding.

Reply:
I wouldn't put the command in the struct, though. (See the existing CLI wrappers)

Reply:
> So this is actually trickier. You can't put the command in the struct, because you can't reuse the command.

That's fair enough - as MB said, you can have a method which returns a new instance of the `Command` every time.

> EDIT: it actually seems like you can set the context per kubectl command. 

Cool so let's do that.

You can [also use the `KUBECONFIG` env variable](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/#set-the-kubeconfig-environment-variable) which helm also respects.

Basically - what I really want to avoid is us misconfiguring peoples' (and our own) `kubectl` client if possible.

If it really isn't - then we need a plan B.

Reply:
> as MB said, you can have a method which returns a new instance of the Command every time.
Sounds good. That makes sense just for kubectl and helm though, because we're passing context to those. The other ones will have no arguments in common, so we can probably just keep them inlined IMHO

In regards to helm, I just saw here (for some reason missed when I looked earlier) that helm has `--kube-context`, so I can just use that for it, then we should be good :)